### PR TITLE
[content]hai-outfits

### PR DESCRIPTION
**Pull request by Wrzlprnft**
# Summary
With this pull i want to replace the images for the hai outfits with more, in my opinion hai-like, images. Since i like the idea of them sharing not only the tier of development but some techniques with the humans, i tried so copy the style of the outfits.
### Todo:
- [x] Add all images
- [x] Add to the datafiles

# Planned outfit images:
- [x] Reactors:
       - [x] Fusion reactor
       - [x] Fission reactor
       - [x] Dwarf core
- [x] Batteries - all sizes
- [x] Atomic thrusters - all sizes
- [x] Atomic steering - all sizes
- [x] Anti-missle turret
- [x] Heavy anti-missle turret

# Data-Code
I will duplicate all the outfits(maybe for further data tweaking for more hai-like efficiency)
they are named "-outfitname- (Hai)"
# Example
### Human fusion reactor:
fusion
### Proposed hai fusion reactor:
fusion hai
# 2x files
https://github.com/endless-sky/endless-sky-high-dpi/pull/8
https://drive.google.com/folderview?id=0B92_vuqiHabWYVlTaU5FRkR3M1k&usp=sharing
- [x] @LocalGod79 hijacking the pull

### CODE REVIEW

**Comment by jafdy** [comment]
Very nice images.
As Hai technology is almost the same as human technology you could just put 'Hai' in front of each outfit. 
With the hyperdrive if you make a Hai version then someone will also need to create a Wanderer and Korath ones as well as a new one for any new species added to the game. I think Hyperdrives need a separate pull.

**Comment by Wrzlprnft** [comment]
it just makes sense, that a so basic outfit like the hyperdirve, which can reverse engineered by every race, has a more "personal" touch by them. ill remove it from this list, though.

About the names, i am not sure, since the grouping in the outfitter and stuff.

**Comment by LocalGod79** [comment]
i believe that your plan for the actual outfit data is a good one; duplicating, adding (Hai) after the parallel outfit name, altering only the sprite attribute - let M.Z. give them the spec flavour that he had intended for the Hai outfits.

i agree that each race should have its own hyperdrive sprite - if for nothing other than immersive depth.  it may be a universal technology, but different species are bound to have slightly different specs (jump speed, scram angle, size, etc.) which might make a universal sprite confusing, and at the very least, it will make the "special" hyperdrives harder to spot at a glance...

**Comment by Wrzlprnft** [comment]
I'll do it the way local suggested.
i think i might get this finished within 24 hours.

and thanks :)
let's put the discussion about hyperdrives aside, at least in this thread.

**Comment by Amazinite** [comment]
These look amazing.

**Comment by Wrzlprnft** [comment]
I think this is, contentwise, finished so far. I may reorganize the hai outfits into one file, additionally. Since most outfits are basically copied from what mr.z did, i am not sure how to handle that.

**Comment by Wrzlprnft** [comment]
Also, the lightning bug and the aphid i think are missing the definition of the turret hardpoint's weapon. not sure if that is on purpose.

**Comment by Wrzlprnft** [comment]
So, im done here, i think. What a day, though. I hope everyone likes those.

**Comment by jafdy** [comment]
6 hours. That is extremely impressive.

**Comment by LocalGod79** [comment]
not to take away from the accomplishment, but i think it's closer to 36 hours - the initial commit was two days ago, the last image 5 hours ago...

@Wrzlprnft just created the p.r. a couple of days after the branch and initial commits...

great job!

**Comment by endless-sky** [comment]
It seems to me that if the Hai are going to have their own distinct outfits, those outfits should have different attributes in addition to different images - like, maybe Hai outfits could be slightly more heat efficient, or have sizes that fill in the gaps in between the sizes of the equivalent human outfits. Otherwise I feel like players are going to be disappointed when they discover what they think is a whole new set of outfits, then find they're just the old outfits re-skinned.

**Comment by Wrzlprnft** [comment]
But it is a better placeholder for now.

**Comment by LocalGod79** [comment]
@endless-sky i believe that @Wrzlprnft was hoping for some direction to assign values for these outfits, and since you seemed to have a plan for everything else, you might have had some advice to keep some balance, in spite of smudging some specs to advantage.

in short - are there any attributes that will specifically "mess with" the Hai ships balance?

**Comment by Amazinite** [comment]
[Hai Tech.txt](https://github.com/endless-sky/endless-sky/files/239656/Hai.Tech.txt)
Here are some different stats that could possibly be used. The descriptions are still the same though.
I made all the hai outfits about 10% to 15% more efficient while also being about 15% to 25% more expensive. This places their technology just above humanity's but not quiet at the level of something like the Wanderers or Korath. I left some comments with the changes and asked a few questions about what might need altering.

**Comment by LocalGod79** [comment]
altering stats like that basically renders the human versions obsolete (and is generally lazy).

the ducts are flat-out broken.

the hai are technologically on the *same tier* as humans - their outfits shouldn't be the same, only just a little better/more expensive.

i've already prepared a set of "in betweens" stats (which in themselves incorporate some of the bonuses you've applied), so that the batteries and engines have tonnage/capacity requirements between the human steps (the last step taking a corresponding step upward).

it'll be in @Wrzlprnft mail box in his morning, and probably the pull by ours...

**Comment by Amazinite** [comment]
Fair enough. I'll take the struggle of trying to find different descriptions for all the engines without them being exactly the same.

**Comment by Alkallid** [comment]
Since the Hai ships don't really have memetical (I think this is the wrong word, they're arbitrary?) outfit, engine, weapon, and cargo spaces, their own tech wouldn't come in sizes that are multiples of 5 or 10, so these could be space-fillers or used for different outfit combos.

**Comment by LocalGod79** [comment]
this is my thought - as long as we're making new outfits (at the atomic engine level), let's add more customizability...

ftr, engines took a half step up, power systems took a half step down - but both got efficiency/price boosts.

**Comment by endless-sky** [comment]
Given how few ships have space for the largest atomics, let alone something "half a step" above that - maybe the engines should go half a step down in size instead of up? Tiny atomic engines could actually be quite useful for many ships.

**Comment by Amazinite** [comment]
I've got some drafted descriptions for the engines (although they refer to the engines being stepped up as opposed to being stepped down). @LocalGod79 where should I send these for now, or will you be changing the engines to be stepped down as @endless-sky suggested?

**Comment by LocalGod79** [comment]
agreed - it was something i thought about when i got to the top...i'll create the engines a step down, but it's getting late, and i think i'll finish it tomorrow morning.

@Amazinite if you wouldn't mind creating a topic in the google forums with an attachment, or even just the text, i'll get it to @Wrzlprnft by his evening productivity spike.

**Comment by Wrzlprnft** [comment]
I just planned to provide different images for the hai, yes kinda reskinned, for someone else to alter the stats, because i am not good at balancing.
Since the hai and the humans are sharing quite a few outfits at the moment, i thought it could be only a step up, even if it is nothing final.
Also, for the player just finding reskinned outfits... maybe something for conspiracy theories or so?

Anyway, main facts: Better than it is at the moment,
Additional facts: Files cleaned up, a base for further work

**Comment by LocalGod79** [comment]
this makes things interesting...

**Comment by Amazinite** [comment]
https://groups.google.com/forum/m/#!category-topic/endless-sky/content-creation/ahrsxgmQ5Ms

The descriptions I came up with. I just copied and pasted the note I had made, so it includes the old descriptions and some of the first descriptions that I came up with for the step up versions.

**Comment by Amazinite** [comment]
The end of the reverse thruster description needs changed. It still says about how they must be placed in the weapons section, but you changed it to require engine space.

**Comment by LocalGod79** [comment]
just need to do new battery descriptions, check the ships outfit space (and make adjustments), and this looks about finished...

**Comment by Amazinite** [comment]
Just to make sure every one sees this. 
Quoted from the discussion post I made that includes the new descriptions. 




12:37 PM
Michael Zahniser
Yes, I think it would make sense to change the names, too. My two goals in naming are:

1. Engines names should be in alphabetical or reverse alphabetical order by size (chipmunk, greyhound, impala..., asteroid, comet, lunar... etc.) so they are sorted from smallest to largest in the outfitter.

2. If possible, names should reflect something of the species that creates the outfit. Wanderers use nature names, for example. Hai names could express solidity, stability, precision, efficiency, durability, etc. in some way, or they could just be random "words" in Hai language (lots of vowels and soft, liquid consonants).

For example, maybe the Hai reactors could be named after different types of stone ("Pebble" Reactor, "Geode" Reactor, "Boulder" Reactor). Maybe battery types could be geological features (chasm, fissure, gorge, ravine, valley). Engine types could be metals: "Aluminum" Class Thruster, bronze, iron, steel, titanium. (Or, maybe you could think of more fitting names.)

**Comment by LocalGod79** [comment]
ah, didn't see that...i went with the insect theme, top 5 fastest flying insects...

i'll come up with something tonight...

**Comment by LocalGod79** [comment]
so, i guess just battery descriptions before this is wrapped up...

**Comment by Amazinite** [comment]
If you need a break, I could try and do some battery descriptions.

**Comment by Wrzlprnft** [comment]
The engine-images will recieve a workover, since they are the least-modificated.

**Comment by LocalGod79** [comment]
descriptions are done (if uncommitted)

working the ships over

**Comment by LocalGod79** [comment]
i think that about does it for the text end of things...

**Comment by LocalGod79** [comment]
so, i kinda hijacked @Wrzlprnft's pull - he lost interest somewhat, when it became more fleshing out the Hai outfit system (which he has little confidence doing) than creating placeholder images for their placeholder outfits.

fortunately, GitHub is pretty flexible...

@endless-sky, input?

File: data/hai outfits.txt [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+	description "This large battery pack boasts higher energy density than smaller models, and helps keep ships fighting in spite of sustaining several Ion Cannon impacts."
+
+outfit "Hai Valley Batteries"
+	category "Power"
+	cost 300000
+	thumbnail "outfit/huge battery hai"
+	"mass" 120
+	"outfit space" -120
+	"energy capacity" 20000
+	description "While most Hai ships do not employ their largest battery model, a few very powerful warships are occasionally equipped with one, keeping their heat footprint low, and in a fight against Ion Cannon equipped opponents through to victory."
```
Comment (toilethinges) [inline data/hai outfits.txt]: and in a fight against Ion Cannon equipped opponents through to victory - this doesn't really make sense if you think about it.

**Comment by Alkallid** [comment]
It uses the 'keeping' from the previous clause to continue, and it only sounds confusing if you overthink it, but yeah, mayne "will surely triumph over" or something.

File: data/weapons.txt [hunk 2]
```diff
@@ -637,6 +566,8 @@ outfit "Javelin Pod"
 		"missile strength" 2
 	description "The Javelin Pod fires a rapid stream of unguided missiles. A Javelin Pod does far more damage than any gun of a comparable size, and has a longer range as well, but once a ship has fired off all its ammunition it must leave combat and find a planet where it can buy more, which makes Javelins less useful in protracted battles. As a result, they are very popular with pirates as a way to quickly disable a small target."
 
+<<<<<<< HEAD
+=======
```
Comment (jmhorjus) [inline data/weapons.txt]: Merge bug.

Comment (Wrzlprnft) [inline data/weapons.txt]: *cough* @LocalGod79 *cough*

**Comment by Wrzlprnft** [comment]
Although, the normal anti-missle might get and overhaul as long as it's open

**Comment by endless-sky** [comment]
Please read the [documentation](https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/#files-field) on the format of the copyright file. It's a machine-readable file, which means the syntax is very strict. in particular:

* You can't have spaces in file names; e.g. "images/outfit/anti-missile hai*" will be interpreted as two separate files, "images/outfit/anti-missile" and "hai*". (The best option at present is to put a ? in place of spaces.)
* You can't use a dagger character for "footnotes"; that will be interpreted as part of the file name. Instead, you should have two separate paragraphs for the files with two different comments.

**Comment by LocalGod79** [comment]
@Wrzlprnft and I discussed this before.  i believe that he'd agree this is probably the best option.

**Comment by Wrzlprnft** [comment]
local, you merged the las changes into my master branch and not the hai-outfits.

**Comment by LocalGod79** [comment]
sorry...i guess we'd better finish this up and then get M.Z.s input again...

i think new shield generator models/names are in the future...i'll try to get to the models this afternoon...

for names, i was thinking "Corundum" to replace S-270 and "Diamond" to replace S-970.  i suspect the alphabetic and hardness scaling order will be too good to pass up (unless he already had something in mind)...

**Comment by LocalGod79** [comment]
i just need to double check the ships to make sure there are no negative outfit space values (and swap some batteries to make space if necessary)

**Comment by Wrzlprnft** [comment]
Can you make a model for the normal anti missle which differs way more from the human one? I didn't make a good version yet.

**Comment by LocalGod79** [comment]
you know the trick i've been using to give outfits a "hai look"...just throw some modified cubes at some intersections...

**Comment by LocalGod79** [comment]
i'm not feeling very well this evening - i don't know if i'm even going to get around to the regenerator sprites...

assets are in the usual places;
https://drive.google.com/folderview?id=0B92_vuqiHabWYVlTaU5FRkR3M1k&usp=sharing

**Comment by toilethinges** [comment]
But... the original regenerators were Hai already, not Syndicate.

**Comment by Alkallid** [comment]
I think it was just convenience; they had similar technologies or just plot holes. However; the Silverhawk had Regenerators, which doesn't fit it being Syndicate tech...

**Comment by LocalGod79** [comment]
@Alkallid what's a Silverhawk?

also, *S*-270 and *S*-970 regenerators are *not S*yndicate creations?  how can you be sure?

i was always under the impression that (aside from their weapons), the Hai were using human technologies as placeholders...

**Comment by Wrzlprnft** [comment]
```
event "syndicate tech available"
	outfitter "Syndicate Basics"
		"Ionic Afterburner"
		"S-270 Regenerator"
	outfitter "Syndicate Advanced"
		"Ionic Afterburner"
		"S-270 Regenerator"
		"S-970 Regenerator"
```

**Comment by LocalGod79** [comment]
it's been my experience that most players don't know what i'm talking about when i say 'checkmate' vs. 'reconciliation'...i think that M.Z. did such a great job building the mystery of "who bombed Geminus/Martini?", that everyone saves the defector to solve it...

**Comment by LocalGod79** [comment]
@Wrzlprnft can we rename this pull request "[content] Hai Outfits"

also, something i that occurred to me during the first rendering of the regenerator sprites, is that this is going to exacerbate #1069

**Comment by jafdy** [comment]
That will happen unless the game stops getting more graphics (ships/effects/outfits) so we may as well make it as good as possible. It could be a "low-dpi-plugin" thing to put Hai outfits back to the old images.

**Comment by toilethinges** [comment]
@LocalGod79 because it is never sold by the Syndicate.

**Comment by LocalGod79** [comment]
i've discussed some options in that thread - seemed a more appropriate location.

**Comment by endless-sky** [comment]
All the game's current outfit images combined take up 12.5 MB of VRAM.

Cutting out the galaxy image and the supernova image saves 67.7 MB of VRAM.

So, adding a few more outfit images is a drop in the bucket compared to that.

**Comment by LocalGod79** [comment]
@Wrzlprnft i think to avoid confusion in the copyright between the heavy and normal anti-missile systems, you did a great job touching that up.  the assets are in the usual place, if you want to play with those "legs" some more...

@endless-sky when reassembling the hai ships, the Shield Beetle was balanced to overheat if exposed to any real external addition - really pushing the limits during combat operation in some cases.  you may wish to do additional outfit refitting, once this is merged.

is there anything about this pull that still stands out to you?

**Comment by Wrzlprnft** [comment]
Hai firing effects still in the same file than the others?

**Comment by Wrzlprnft** [comment]
Local and i agreed that this one is finished so far, now it's up to you, @endless-sky . You can give us further input, and we'll work on it, or merge it (or trash it, but please don't do that).

**Comment by LocalGod79** [comment]
sorry about the artifacting, but i wanted to keep the filesize down
haioutfits

**Comment by LocalGod79** [comment]
haioutfits copy

**Comment by Alkallid** [comment]
Um... What about that Supercapacitor?

**Comment by LocalGod79** [comment]
it's not on any Hai ships.  maybe it gets removed from the Hai outfitters, maybe it's a generic/small enough part that the humans could have brought enough of them to stock at their outfitters (and the Hai are hoping you'll just buy them, so they'll have more inventory space)...

**Comment by LocalGod79** [comment]
we'll just have to wait to see what @endless-sky thinks.

**Comment by endless-sky** [comment]
It looks like the A1 thruster is more efficient (in terms of energy, heat, and space) than the A2 thruster, which is probably a mistake. Also, the A3 steering has most of the same attributes as the human medium atomic steering, rather than being halfway between two human atomic levels like the rest of the engines.

The spurious changes in outfits.txt and power.txt should probably be fixed, too. (For tracking the change history of files, it's just useful if there aren't extra changes listed that are not actually doing anything.)

I do really want to get this merged in before 0.9.1 (i.e. in the next week and a half or so), since this is the sort of big balance change that's more suited to an "unstable" release. (My hope is for 0.9.2 to be a "stable" release just focused on bug fixes and smaller tweaks.)

**Comment by LocalGod79** [comment]
i see no significant obstacles to getting this merged on the 0.9.1 timeline, if those are your only concerns - i've already begun addressing them locally.

**Comment by LocalGod79** [comment]
i think i'm getting a better handle on commits, and those address the concerns you raised.

the first time around, i duplicated the human outfits and tried to create a modified value to suit the need.  this time, i created a spreadsheet, and had it run the math down the table, and i plugged in the new values.

it revealed a couple of other issues as well (e.g. the Hai A2 steering having the same heat as the Human A1 steering, as well as other scaling issues - apparent in that commit).

**Comment by endless-sky** [comment]
It would also make sense to take out the "A1", "A2" etc. from the thruster and steering model numbers. Presumably the Hai have their own alphabet where the letter that makes the "A" sound is something other than A, in the same way that the Korath have Kar, Lek, Met, etc. instead of Kay, El, Em, etc. And the Hai probably aren't big on assigning numbers to things in place of names, anyway.

**Comment by LocalGod79** [comment]
I've noticed that *all* the other alien engines start with the same letter (which groups them in the outfitters)

was there anywhere in the human alphabet you'd like the Hai engines to fall? (i have a system in mind)

**Comment by LocalGod79** [comment]
perhaps this could be as simple as explaining that "Human nomenclature is a rough translation from the Hai version, which otherwise seems nonsense." in the A1 steering/thruster descriptions?

**Comment by endless-sky** [comment]
Well, it could be as simple as just calling them "Hai Butterfly Thruster," "Hai Cleg Thruster," etc. That would group them *and* make it clear whose technology they are, which would match what you did with the batteries, cooling, and shields.

**Comment by LocalGod79** [comment]
but, it wouldn't retain the order that you desire for scaling outfits (as the names were chosen for their abstract properties, rather than alphabetic)

**Comment by endless-sky** [comment]
Oh. I saw that they were partly alphabetical (except horsefly) and assumed they were all alphabetical.

**Comment by LocalGod79** [comment]
many of the names were truncated to fit the outfitter screen, and (upon revision and as a happy accident) appear to have fallen into a nearly completely alphabetic order.

the names were chosen as the "five fastest flying insects", and over time, i've grown to feel like blurring the line between the engines and ship nomenclature is probably not a great idea.

**Comment by LocalGod79** [comment]
okay, so here are some made up names that might fit the Hai linguistic style, and offers nomenclature according to standard:

`outfit "'Lasrem' Atomic Thruster"
outfit "'Lenga' Atomic Thruster"
outfit "'Liroo' Atomic Thruster"
outfit "'Londir' Atomic Thruster"
outfit "'Lums' Atomic Thruster"`

`outfit "'Lasrem' Atomic Steering"
outfit "'Lenga' Atomic Steering"
outfit "'Liroo' Atomic Steering"
outfit "'Londir' Atomic Steering"
outfit "'Lums' Atomic Steering"`

obviously, the only thing set in stone is the *second* letter, and that the first letter needs to be uniform.

**Comment by endless-sky** [comment]
That could work, but for consistency please use double quotes (enclosing the entire outfit name in backticks) rather than single quotes.

**Comment by LocalGod79** [comment]
would you prefer 'S' instead of 'L', so that it doesn't split the Plasma engines up any further?  or 'B'?

**Comment by LocalGod79** [comment]
got hasty with the commits again - i'll do better going forward, i promise.

otherwise, i believe this is ready...

**Comment by Amazinite** [comment]
Benga steering has an extra comma in the last sentence.

**Comment by LocalGod79** [comment]
@endless-sky changelog is updated.  looks good to go.

**Comment by endless-sky** [comment]
@LocalGod79, can you send me the @2x versions of these outfit images? You sent me some of them, but I'm missing the assets for the regenerators, and I also don't know if any of the images have changed since you sent them to me (a month ago). Thanks!

**Comment by LocalGod79** [comment]
@endless-sky updated address for assets pertaining to this pull;
https://drive.google.com/file/d/0B92_vuqiHabWSDFScDZOal8xdG8/view?usp=sharing

if you would like, it probably wouldn't be too difficult to merge the GIMP assets with the human placeholders they "replaced" - they're by far the largest portion of the data, and have only a layer or two different than the human versions.

**Pull request merged**